lib/puppet/module_tool/applications/builder.rb

```diff
@@ -40,6 +41,22 @@ def archive_file
         File.join(@pkg_path, "#{metadata.release_name}.tar.gz")
       end
 
+      def minitar_verify
+        begin
+          # assume valid if minitar inaccessible (i.e. immediate rescue loading gem)
+          require 'archive/tar/minitar'
+        rescue
+          return true
+        end
+
+        # minitar present, so assume invalid
+        files = Puppet::ModuleTool::Tar::Mini.new.entries(archive_file)
+        valid = files.all? { |f| f.full_name == f.name }
```

Comment:
Not even positive this comparison is accurate - on OSX, when run with DSC module, the entries contain a bunch of oddly encoded strings.  Not sure if minitar just doesn't work right on OSX?